File: food-backend/src/models/cart.model.ts
import { Schema, model, Document, Types } from "mongoose";
import { Carts } from "../types/cart.d";
import { Users } from "../types/user.d";
import { MenuItemType } from "../types/menu.d";

const cartSchema = new Schema<Carts>(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    items: [
      {
        menuItemId: {
          type: Schema.Types.ObjectId,
          ref: "MenuItem",
          required: true,
        },
        quantity: { type: Number, required: true, min: 1 },
      },
    ],
  },
  { timestamps: true }
);

export const Cart = model<Carts>("Cart", cartSchema);

File: lib/models/cart_model.dart
import 'menu_item.dart';
class Cart {
  final String id;
  final String userId;
  final List<CartItem> items;

  Cart({
    required this.id,
    required this.userId,
    required this.items,
  });

  factory Cart.fromJson(Map<String, dynamic> json) => Cart(
        id: json['_id'],
        userId: json['userId'],
        items: (json['items'] as List)
            .map((item) => CartItem.fromJson(item))
            .toList(),
      );
}

class CartItem {
  final MenuItem menuItemId;
  final int quantity;

  CartItem({
    required this.menuItemId,
    required this.quantity,
  });

  factory CartItem.fromJson(Map<String, dynamic> json) => CartItem(
        menuItemId: MenuItem.fromJson(json['menuItemId']),
        quantity: json['quantity'],
      );
}